WS2012R2/lisa/stateEngine.ps1

```diff
@@ -817,49 +824,159 @@ function DoSystemDown([System.Xml.XmlElement] $vm, [XML] $xmlData)
         }
         else
         {
-            $testData = GetTestData $vm.currentTest $xmlData
-            if ($testData -is [System.Xml.XmlElement])
+            UpdateState $vm $ApplyCheckpoint
+        }
+    }
+    else
+    {
+        #for NON-SUT VMs, put the VM state to run preStartScript or StartSystem
+        if ($vm.preStartConfig)
+        {
+            UpdateState $vm $RunSetupScript
+        }
+        else
+        {
+            UpdateState $vm $StartSystem
+        }
+    }
+}
+
+
+########################################################################
+#
+# DoApplyCheckpoint()
+#
+########################################################################
+function DoApplyCheckpoint([System.Xml.XmlElement] $vm, [XML] $xmlData)
+{
+    <#
+    .Synopsis
+        Apply checkpoint to let VM go into a know status
+        if RevertDefaultSnapshot=True.
+    .Description
+        Apply checkpoint if RevertDefaultSnapshot=True. Then transition 
+        to RunSetupScript if the currentTest defines a setup script.
+        Otherwise, transition to StartSystem. If RevertDefaultSnapshot=False
+        or not configured, there's no checkpoint restoring applied in VM
+        and do state transition instead.
+    .Parameter vm
+        XML Element representing the VM under test.
+    .Parameter xmlData
+        XML document for the test.
+    .Example
+        DoSystemDown $testVM $xmlData
+    #>
+
+    if (-not $vm -or $vm -isnot [System.Xml.XmlElement])
+    {
+        LogMsg 0 "Error: DoApplyCheckpoint received an bad VM parameter"
+        return
+    }
+
+    LogMsg 9 "Info : Entering DoApplyCheckpoint( $($vm.vmName) )"
+
+    if (-not $xmlData -or $xmlData -isnot [XML])
+    {
+        LogMsg 0 "Error: DoApplyCheckpoint received a null or bad xmlData parameter - VM $($vm.vmName) disabled"
+        $vm.emailSummary += "    DoApplyCheckpoint received a null xmlData parameter - VM disabled<br />"
+        $vm.currentTest = "done"
+        UpdateState $vm $Disabled
+    }
+
+    $testData = GetTestData $vm.currentTest $xmlData
+    if ($testData -is [System.Xml.XmlElement])
+    {
+        # Do not need to recover from checkpoint
+        if (-not $testData.RevertDefaultSnapshot -or $testData.RevertDefaultSnapshot -eq "False")
+        {
+            LogMsg 9 "Info : noCheckpoint is not configured or set to True."
```

Comment:
Update the log message so that it mentions "RevertDefaultSnapshot" instead of "noCheckpoint"